compiler/llvm/clangUtil.cpp

```diff
@@ -2453,6 +2463,25 @@ void configurePMBuilder(PassManagerBuilder &PMBuilder, bool forFunctionPasses, i
 
 static void registerDumpIrExtensions(PassBuilder& PB);
 
+// expose a command line option from LLVM
+// we set it via a chpl command line option
+namespace llvm {
+extern cl::opt<bool> PrintPipelinePasses;
+}
+static PassBuilder constructPassBuilder(llvm::TargetMachine* targetMachine, PassInstrumentationCallbacks* PIC, bool forFunction) {
+  // this is required to be set, or LLVM will not properly populate the pass
+  // names. technically this flag enables extra printing to the dbg() output,
+  // but we only keep the flag long enough to populate the pass names.
+  // this must always be set so that `--print-before` (and similar commands) will
+  // still work with their nice pass name
+  llvm::PrintPipelinePasses = true;
```

Comment:
Shouldn't adjusting this global be inside `if (fLlvmPrintPasses)` here and three lines later?

Reply:
It has to always be set or LLVM won't populate pass names. This is so `--mllvm --print-before=mem2reg` works, otherwise it uses the class name of the pass. Which LLVM won't ever print out, you can only know it by inspecting the source code.

Reply:
OK, thanks for explaining. I'd expect this is some kind of performance optimization on their part, though? I am not imagining it will be significant for us.